idaes/models/properties/modular_properties/eos/ceos.py

```diff
@@ -901,6 +902,125 @@ def vol_mol_phase_comp(b, p, j):
+        ctype = pobj._cubic_type
+
+        # Add components for calculation of mixture critical properties
+        def func_a_crit(m, j):
+            cobj = m.params.get_component(j)
+            fw = getattr(m, cname + "_fw")
+            return (
+                EoS_param[ctype]["omegaA"]
+                * (
+                    (Cubic.gas_constant(m) * cobj.temperature_crit) ** 2
+                    / cobj.pressure_crit
+                )
+                * (
+                    (1 + fw[j] * (1 - sqrt(m.temperature_crit / cobj.temperature_crit)))
+                    ** 2
+                )
+            )
+
+        m.add_component(
+            "a_crit",
+            Expression(
+                m.component_list, rule=func_a_crit, doc="Component a coefficient"
+            ),
+        )
+
+        def rule_am_crit(m):
+            try:
+                rule = m.params.get_phase(ref_phase).config.equation_of_state_options[
+                    "mixing_rule_a"
+                ]
+            except (KeyError, TypeError):
+                rule = MixingRuleA.default
+
+            a_crit = getattr(m, "a_crit")
+            if rule == MixingRuleA.default:
+                return rule_am_crit_default(m, cname, a_crit)
+            else:
+                raise ConfigurationError(
+                    "{} Unrecognized option for Equation of State "
+                    "mixing_rule_a: {}. Must be an instance of MixingRuleA "
+                    "Enum.".format(m.name, rule)
+                )
+
+        m.add_component("am_crit", Expression(rule=rule_am_crit))
+
+        def rule_bm_crit(m):
+            try:
+                rule = m.params.get_phase(ref_phase).config.equation_of_state_options[
+                    "mixing_rule_b"
+                ]
+            except KeyError:
+                rule = MixingRuleB.default
+
+            b = getattr(m, cname + "_b")
+            if rule == MixingRuleB.default:
+                return rule_bm_crit_default(m, b)
+            else:
+                raise ConfigurationError(
+                    "{} Unrecognized option for Equation of State "
+                    "mixing_rule_b: {}. Must be an instance of MixingRuleB "
+                    "Enum.".format(m.name, rule)
+                )
+
+        m.add_component("bm_crit", Expression(rule=rule_bm_crit))
+
+        def rule_A_crit(m):
+            am_crit = getattr(m, "am_crit")
+            return EoS_param[ctype]["omegaA"] * (
+                Cubic.gas_constant(m) * m.temperature_crit
+            ) ** 2 == (am_crit * m.pressure_crit)
+
+        m.add_component("A_crit", Constraint(rule=rule_A_crit))
+
+        def rule_B_crit(m):
+            bm_crit = getattr(m, "bm_crit")
+            return EoS_param[ctype]["coeff_b"] * Cubic.gas_constant(
+                m
+            ) * m.temperature_crit == (bm_crit * m.pressure_crit)
+
+        m.add_component("B_crit", Constraint(rule=rule_B_crit))
```

Comment:
Can't these be `Expressions`? Or is the idea to avoid divisions.

Reply:
No, these are the actual constraints we need to solve to find T_c and P_c.